app/src/main/java/com/alphawallet/app/entity/TransactionContract.java

```diff
@@ -176,8 +176,8 @@ public String getOperationResult(Token token, TransactionOperation operation, Tr
             if (!token.isNonFungible() && (operation.value == null || operation.value.length() > 0 && Character.isDigit(operation.value.charAt(0))))
             {
                 BigDecimal value = new BigDecimal(operation.value);
-                //appears to be a number; try to produce number with prefix
-                return tx.getPrefix(token) + BalanceUtils.getScaledValueFixed(value, token.tokenInfo.decimals, TransactionHolder.TRANSACTION_BALANCE_PRECISION);
+                if (getOperationId() == TransactionType.APPROVE.ordinal() && value.abs().compareTo(token.balance.multiply(BigDecimal.TEN)) > 0) return "All";
```

Comment:
Compare to 0xFFF....FFF here instead for simplicity

Reply:
Sometimes the approve amount is not BigInteger(-1), in which case this won't work.

I've seen a few dapps use a different, large amount.